spec/lib/cronofy/client_spec.rb

```diff
@@ -2690,4 +2690,145 @@
       end
     end
   end
+
+  describe '#create_scheduling_conversation' do
+    let(:request_url) { 'https://api.cronofy.com/v1/scheduling_conversations' }
+    let(:method) { :post }
+    let(:request_body) do
+      {
+        "participants" => [
+          {
+            "participant_id" => "@grace",
+            "sub" => "acc_567236000909002",
+            "slots" => {
+              "choice_method" => "auto"
+            }
+          },
+          {
+            "participant_id" => "@karl"
+          }
+        ],
+        "required_duration" => { "minutes" => 60 },
+        "available_periods" => [
+          {
+            "start" => "2018-05-01T00:00:00Z",
+            "end" => "2018-05-08T23:59:59Z"
+          }
+        ]
+      }
+    end
+
+    let(:correct_response_code) { 200 }
+    let(:correct_response_body) do
+      {
+        "scheduling_conversation" => {
+          "scheduling_conversation_id" => "abcd1234"
+        }
+      }
+    end
+
+    let(:correct_mapped_result) do
+      Cronofy::SchedulingConversation.new(scheduling_conversation_id: "abcd1234")
+    end
+
+    subject { client.create_scheduling_conversation(request_body) }
+
+    it_behaves_like 'a Cronofy request'
+    it_behaves_like 'a Cronofy request with mapped return value'
+  end
+
+  describe '#lookup_scheduling_conversation' do
+    let(:token) { "hagsdau7g3d" }
+    let(:request_url) { "https://api.cronofy.com/v1/scheduling_conversations?token=#{token}" }
+    let(:method) { :get }
+
+    let(:correct_response_code) { 200 }
+    let(:correct_response_body) do
+      {
+        "participant" => {
+          "participant_id" => "83o38hoa"
+        },
+        "scheduling_conversation" => {
+          "scheduling_conversation_id" => "abcd1234"
+        },
+      }
+    end
+
+    let(:correct_mapped_result) do
+      Cronofy::SchedulingConversationResponse.new(
+        participant: Cronofy::Participant.new(participant_id: "83o38hoa"),
+        scheduling_conversation: Cronofy::SchedulingConversation.new(scheduling_conversation_id: "abcd1234"),
+        )
+    end
+
+    subject { client.lookup_scheduling_conversation(token) }
+
+    it_behaves_like 'a Cronofy request'
+    it_behaves_like 'a Cronofy request with mapped return value'
+  end
+
+  describe '#upsert_availability_rule' do
```

Comment:
Should have specs for get and delete too.

Reply:
Done